src/session_state.rs

```diff
@@ -74,16 +75,34 @@ impl<C: Clock> SessionState<C> {
         }
     }
 
+    pub fn handle_connection_reset(&mut self) {
```

Comment:
Since this is called on `States::ConnectTransport`, shouldn't `last_transmission` and `last_reception` be set to `now`?
Otherwise, on paper, if there is never anything sent and received, `ping_is_due` will not fire and the connection will stay.

OTOH: If, as you may say, you guarantee, that you send at least the connect command, then at least the last_transmission timer will fire and the ping response not coming in would ultimately trigger disconnect.

Either way, let's describe what's supposed to happen in the pathological case where both timestamps remain `None` or why that case doesn't exist.

Reply:
I've refactored sot hat the last_transmission and last_reception will default to indicate timeout if no timestamp is present, which should resolve any issues here. I've also refactored state management such that these conditions should still nominally never happen, so it's a backstop against future bugs.